rotkehlchen/chain/gnosis/modules/sdai/decoder.py

```diff
@@ -0,0 +1,109 @@
+from typing import TYPE_CHECKING, Any
+
+from rotkehlchen.chain.ethereum.utils import token_normalized_value_decimals
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.gnosis.modules.sdai.constants import (
+    CPT_SDAI,
+    GNOSIS_SDAI_ADDRESS,
+    SDAI_ICON,
+    SDAI_LABEL,
+)
+from rotkehlchen.constants.assets import A_SDAI
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.types import ChecksumEvmAddress
+from rotkehlchen.utils.misc import hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.gnosis.node_inquirer import GnosisInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+SDAI_DEPOSIT = b'\xdc\xbc\x1c\x05$\x0f1\xff:\xd0g\xef\x1e\xe3\\\xe4\x99wbu.:\tR\x84uED\xf4\xc7\t\xd7'  # noqa: E501
+SDAI_REDEEM = b'\xfb\xdey} \x1ch\x1b\x91\x05e)\x11\x9e\x0b\x02@|{\xb9jJ,u\xc0\x1f\xc9fr2\xc8\xdb'
+
+
+class SdaiDecoder(DecoderInterface):
+    def __init__(
+            self,
+            gnosis_inquirer: 'GnosisInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=gnosis_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.sdai = A_SDAI.resolve_to_evm_token()
+
+    def _decode_sdai_deposit_events(self, context: DecoderContext) -> DecodingOutput:
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        amount = token_normalized_value_decimals(amount_raw, None)
+        shares_raw = hex_or_bytes_to_int(context.tx_log.data[32:])
```

Comment:
be explicit here with the ranges

```suggestion
        shares_raw = hex_or_bytes_to_int(context.tx_log.data[32:64])
```